### ISSUE TITLE
Provide a way to sanitize third party input in flux queries

### ISSUE THREAD

**Issue by WolfspiritM** [influxdata/influxdb-client-csharp#129]
__Proposal:__
All the examples assume the query is fixed but that's not always the case. In my case I want to let the user specify the time range and variables to search for. For example:

``` c#
            string fluxQuery = $@"
            from(bucket: ""data"")
                |> range(start: {start.ToUnixTimeSeconds()}, stop: {end.ToUnixTimeSeconds()})
                |> pivot(rowKey:[""_time""], columnKey: [""_field""], valueColumn: ""_value"")
                |> filter( fn: (r) => r.device == ""{request.Device}"")";
```

I can't find any way to sanitize data for usage inside the flux query string. What is the best way to make sure the query isn't changed by third party?

__Current behavior:__
 The way it is right now everyone can use ```" and ...``` as device to change the query.
Is it enough to just escape `"` with `\"`?
I think I also need to escape `\` with `\\`?
What else could break out of there?

__Desired behavior:__
Provide a way to securely use c# variables inside flux queries. Maybe with functions that escape a string so it can be used inside a query.

__Alternatives considered:__
Maybe variables can somehow be passed to the query in a secure way.
``` c#
            var variables = new {
                Start = start.ToUnixTimeSeconds(),
                Stop = end.ToUnixTimeSeconds(),
                Device = request.Device
            };
            string fluxQuery = @"
            from(bucket: ""data"")
                |> range(start: Start, stop: Stop)
                |> pivot(rowKey:[""_time""], columnKey: [""_field""], valueColumn: ""_value"")
                |> filter( fn: (r) => r.device == Device)";

           fluxClient.QueryAsync(fluxQuery, variables,... )
```

__Use case:__
Prevents "Flux Injection" attacks.

**Comment by bednar** [comment]
Hi @WolfspiritM,

The API for v2 doesn't have bind (query) parameters - see influxdata/influxdb#16109. 

Our API allows to specify `extern` via [Query](https://github.com/influxdata/influxdb-client-csharp/blob/7cd6a38f6a963db59729fa8116e9f916fcfd46ab/Client/InfluxDB.Client.Api/Domain/Query.cs#L87), but working with `extern` is too painful and doesn't offer an alternative for the unsupported bind parameters.

Regards

**Comment by sranka** [comment]
https://github.com/influxdata/influxdb-client-js/blob/master/packages/core/src/query/flux.ts#L40-L99 sanitizes a string value, you might try porting this to C# to match your needs.

**Comment by bednar** [comment]
Hi @WolfspiritM, we just started works on LINQ interfaces. You could track progress in #146

**Comment by bednar** [comment]
Closed this by #146
